torch/csrc/dynamo/guards.cpp

```diff
@@ -1130,6 +1130,63 @@ class NO_TENSOR_ALIASING : public RelationalGuard {
   long unsigned int _counter = 0;
 };
 
+class DYNAMIC_INDICES : public LeafGuard {
+  // C++ equivalent of
+  // if hasattr(value, "_dynamo_dynamic_indices"):
+  //     code.append(
+  //         f"(({tensor_name}._dynamo_dynamic_indices.issubset({value._dynamo_dynamic_indices}))
+  //         if hasattr({tensor_name}, '_dynamo_dynamic_indices') else True)"  #
+  //         noqa: B950
+  //     )
+  // else:
+  //     code.append(
+  //         f"hasattr({tensor_name}, '_dynamo_dynamic_indices') == False"
+  //     )
+ public:
+  DYNAMIC_INDICES(
+      bool has_attr,
+      py::set dynamic_indices,
+      py::object verbose_code_parts)
+      : LeafGuard(verbose_code_parts),
+        _has_attr(has_attr),
+        _dynamic_indices(dynamic_indices) {}
+
+  bool check_nopybind(PyObject* value) override { // borrowed ref
+    // Make an interned string
+    static PyObject* dynamic_indices_str =
+        PyUnicode_InternFromString("_dynamo_dynamic_indices");
+
+    bool has_attr = PyObject_HasAttr(value, dynamic_indices_str);
+    // Common case - hasattr({tensor_name}, '_dynamo_dynamic_indices') == False
+    if (!_has_attr) {
+      return !has_attr;
+    }
+
+    // "((x._dynamo_dynamic_indices.issubset({value._dynamo_dynamic_indices}))
+    //       if hasattr(x, '_dynamo_dynamic_indices') else True)
+    if (!has_attr) {
+      return true;
+    }
+
+    PyObject* indices = PyObject_GetAttr(value, dynamic_indices_str); // new ref
+    static PyObject* issubset_str = PyUnicode_InternFromString("issubset");
+    PyObject* issubset = PyObject_GetAttr(indices, issubset_str); // new ref
+    PyObject* call_result =
+        PyObject_CallOneArg(issubset, _dynamic_indices.ptr()); // new ref
```

Comment:
Use `PyObject_CallMethodOneArg`.  I believe we have a 3.8 back port of that in one of the compatibility headers (for py38).

Reply:
Does it return a borrowed ref or a new ref? In the latest commit, I assumed its a new ref.

Reply:
New reference.